pkg/route/ingress/metrics.go

```diff
@@ -0,0 +1,102 @@
+package ingress
+
+import (
+	"k8s.io/apimachinery/pkg/labels"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+
+	"github.com/blang/semver/v4"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	routeController                 = "openshift_ingress_to_route_controller"
+	metricUnmanagedRoutes           = routeController + "_route_unmanaged"
+	metricIngressesWithoutClassName = routeController + "_ingress_without_class_name"
+)
+
+var (
+	unmanagedRoutes = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: metricUnmanagedRoutes,
+		Help: "Report the number of unmanaged routes.",
+	}, []string{"name", "host"})
+
+	ingressesWithoutClassName = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: metricIngressesWithoutClassName,
+		Help: "Report the number of ingresses that do not specify ingressClassName.",
+	}, []string{"name"})
+)
+
+func (c *Controller) Create(v *semver.Version) bool {
+	c.metricsCreateOnce.Do(func() {
+		c.metricsCreateLock.Lock()
+		defer c.metricsCreateLock.Unlock()
+		c.metricsCreated = true
+	})
+	return c.MetricsCreated()
+}
```

Comment:
It doesn't look lik `v` is used in the function body. What's the use case for making it a method argument?